management_app/views/faculty.py

```diff
@@ -1,42 +1,221 @@
-from flask import Flask, Blueprint, send_from_directory, render_template, request, redirect, url_for
+import pandas as pd
+import traceback
+from flask import Blueprint, render_template, request, redirect, url_for
 from werkzeug.utils import secure_filename
 
+from management_app.db import get_db
 from management_app.views.auth import login_required
-from management_app.views.utils import download_file, upload_file, remove_upload_file
+from management_app.views.utils import download_file, upload_file, remove_upload_file, get_upload_filepath
 
 faculty = Blueprint('faculty', __name__, url_prefix='/faculty')
 
-@faculty.route('/')
+@faculty.route('/', methods=['GET', 'POST'])
 @login_required
 def index():
-    # TODO: Get data from database
-    faculties = [
-        {'name': 'Professor Mark', 'email': 'mark@uci.edu', 'role': 'Tenured Faculty', 'required_point': 3.5,
-         'prev_balance': 0.1250, 'estimated_balance': 0.2500},
-        {'name': 'Professor 2', 'email': 'mark1@uci.edu', 'role': 'Pre Tenured Faculty 1 st', 'required_point': 2.5,
-         'prev_balance': 0.1000, 'estimated_balance': 0.1250},
-        {'name': 'Professor 3', 'email': 'mark3@uci.edu', 'role': 'PoT Tenured Faculty', 'required_point': 6.5,
-         'prev_balance': 1.4375, 'estimated_balance': 0.9375}
-    ]
-    return render_template('faculty/index.html', faculties=faculties)
-
-
-@faculty.route('/data-templates/<filename>')
+    years = ['2020-2021', '2019-2020']
+    faculties = get_professor_point_by_year(2020) # temporarily set current year 2020
+    return render_template('faculty/index.html', faculties=faculties, years=years)
+
+@faculty.route('/data-templates/<filename>', methods=['GET'])
 @login_required
 def download_template(filename):
-    return download_file(filename)
-
+    if request.method == 'GET':
+        return download_file(filename)
 
 @faculty.route('/upload', methods=['POST'])
 @login_required
-def upload_user_template():
-    if (request.method == 'POST'):
+def upload():
+    if request.method == 'POST':
         file = request.files['facultyTemplate']
         filename = secure_filename(file.filename)
+        file_path = get_upload_filepath(filename)
+        action_type = request.form['templateUploadOptions']
+
         upload_file(file)
 
-        # TODO: process spreadsheet and store in database
+        if action_type == 'addUsers':
+            process_user_template(file_path, 1, 'Users', action_type)
+        elif action_type == 'editUsers':
+            process_user_template(file_path, 1, 'Users', action_type)
+        elif action_type == 'addProfessors':
+            process_professors_point_template(file_path, 1, 'Professors_point_info', action_type)
+        elif action_type == 'editProfessors':
+            process_professors_point_template(file_path, 1, 'Professors_point_info', action_type)
```

Comment:
a little naming suggestion: "template" is the thing users download, would it be better if we change the name to process_user_file to better represent this is about uploading real data

Reply:
Agree. Updated to `process_user_file` and `process_professors_point_file`.